### ISSUE TITLE
update offer should recalculate values

### ISSUE THREAD

**Issue by JMurphyWeb** [karmaradio/karma#116]
a valid update to an offer should re-calculate fields

**Comment by JMurphyWeb** [comment]
linked to #12

**Comment by JMurphyWeb** [comment]
I was struggling to test the final make/update offer changeset. Seems like the initial validation covers everything necessary and I couldn't mock the `run_calculations` function

**Comment by JMurphyWeb** [comment]
For now, a PM cannot update the email address of the contractor. It adds an unnecessary level of complexity at this stage. Field will be un-editable...  @jonscottfilm wanted to bring this to your attention.

Please also get testing the calculated values (update offer and create offer). They can be seen by clicking on an offer. Should probably @beccacca here as well?

**Comment by jonscottfilm** [comment]
Becca talked in #general on Slack today with JC re calculated value errors. @beccacca had some questions.

**Comment by jonscottfilm** [comment]
Ok, so once offer is sent to an email once, email address can't be changed, even if it is a typo... Ok.

**Comment by jackcarlisle** [comment]
@jonscottfilm the main thinking behind this was:
 
If a PM makes an offer to the wrong email address, the initial email will be sent to that wrong email. Then if you update the email, the person with the correct email address will receive an email saying "your offer has been updated" but in their eyes it's the first time they're seeing it.

**Comment by jonscottfilm** [comment]
Gotya. Although it does mean the PM has to re-key the deal info. Perhaps later on we can implement code/functionality that if/when an email address is changed in 'edit offer' it determines that it must therefore need to be the initial sending of the offer. _However_, I suppose the contractor could just have changed email addresses between initial offer and amended offer... 
Perhaps a choice field: 'Initial OR Amended offer?'

Or, when we implement draft offers, and ability to copy/replicate existing offers to use as templates..?

However, when we have SMS 2FA in place, will this issue be avoided? Not sure it will, because intended user will still not receive the initial email. Thinking.

**Comment by jackcarlisle** [comment]
Yeah we can add that functionality in at a slightly later stage. Would you mind opening up a new issue regarding it? I think this is a more viable solution initially than the drafts but again that is something that we'd like to build once the application is more complete

**Comment by jonscottfilm** [comment]
Sure, it's def a later issue. I'll create.

**Comment by JMurphyWeb** [comment]
@jonscottfilm can this be closed? 
if it has no please test and no bug label  ==> we are confused.

**Comment by jonscottfilm** [comment]
Sorry, bad admin. Becca and I still checking calcs in create offer and amend offer.
